Lib/idlelib/pyshell.py

```diff
@@ -387,6 +387,16 @@ def handle_EOF(self):
         "Override the base class - just re-raise EOFError"
         raise EOFError
 
+def restart_line(width, filename):  # See bpo-38141.
+    if filename == '':
+        filename = 'Shell'
+    tag = f"= RESTART: {filename} ="
+    if width >= 13 + len(filename):  # The length of tag.
+        div, mod = divmod((width -len(tag)), 2)
+        return f"\n{(div+mod)*'='}{tag}{div*'='}"
```

Comment:
For better separation of concerns, IMO this function should return the line without a `'\n'` at the beginning.

Reply:
Agreed.  I should not worry about the extra microseconds of 2 writes.  The tests will be clearer also.

Reply:
Beside which, '\n' is usually needed to end a current line with an unused '>>> ' prompt.  We could instead have removed the prompt, or for 'Restart Shell', inserted the restart line above the prompt line.   In any case, we will have to reexamine what to do when the sidebar is added (the next thing to review).